### PR TITLE
Hicexplorer 3.7.6

### PR DESCRIPTION
**Pull request by joachimwolff**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [x] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

Update of HiCExplorer, new tool to build micro-C matrices.

### CODE REVIEW

File: tools/hicexplorer/hicBuildMatrixMicroC.xml [hunk 1]
```diff
@@ -0,0 +1,225 @@
+<tool id="hicexplorer_hicbuildmatrixmicroc" name="@BINARY@" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@">
```
Comment (bgruening) [inline tools/hicexplorer/hicBuildMatrixMicroC.xml:1]:
```suggestion
<tool id="hicexplorer_hicbuildmatrixmicroc" name="@BINARY@" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
```

Can you please add this to all tools please?

File: tools/hicexplorer/hicBuildMatrixMicroC.xml [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    <description>create a contact matrix</description>
+    <macros>
+        <token name="@BINARY@">hicBuildMatrixMicroC</token>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements">
+        <requirement type="package" version="1.19">samtools</requirement>
```
Comment (bgruening) [inline tools/hicexplorer/hicBuildMatrixMicroC.xml]:
```suggestion
        <requirement type="package" version="1.21">samtools</requirement>
```

**Review by bgruening — COMMENTED** [review]
Hi @joachimwolff nice to see you here again.

I hope bumping the samtools requirement will fix the error.

File: tools/hicexplorer/hicBuildMatrixMicroC.xml (hunk 1 above)
Comment (joachimwolff) [reply tools/hicexplorer/hicBuildMatrixMicroC.xml]: Was already there for the most tools via macros.xml. Added it where it was missing.

**Comment by joachimwolff** [comment]
I don't know why it fails here, on the local machine of mine everything is passing. Is there a way to get the tool_test_output.html? I searched in the logs for a hint what is wrong, but don't see something obvious.

**Comment by bgruening** [comment]
You will find the html on this site https://github.com/galaxyproject/tools-iuc/actions/runs/12134110993?pr=6595

grafik


Unfortunately, HiCE is not providing any useful error message as it seems. Can it be that it uses too much memory?
Please also make sure you are running planmoe with `--biocontainers`, we do this here on CI as well.

**Comment by joachimwolff** [comment]
Is it possible to increase the memory?

**Comment by bgruening** [comment]
In GitHub Actions, I don't think so. But it should have 16GB? How much do you need?

**Comment by joachimwolff** [comment]
16 GB is way more than it should need, this is the usual toy data we use since at least 2016. The data was not changed, only the organisation of HiCExplorer in the background and the micro-C part drops a few parameters in comparison to hicBuildMatrix. This should all run through as always.

**Comment by bgruening** [comment]
Maybe you understand the error better than me. Memory was just an assumption. I will also clone it and run it locally later this day.

**Comment by joachimwolff** [comment]
Using --biocontainers leads to a fail of all tools on my local machine, but only if a subset is run. A run of all leads not to a crash, but at some point it simply hangs and does not continue to test. Unclear to me what is happening here.

**Comment by joachimwolff** [comment]
I got it locally with biocontainers running, and the hicBuildMatrix / hicBuildMatrixMicroC test cases were failing. The only difference I saw between the passing tools and the hicBuildMatrix / hicBuildMatrixMicroC was the additional requirement to samtools. After removal, the tests are passing.

Given that HiCExplorer installs via conda dependency resolving a samtools version, I am not sure why we need to declare this dependency explicit in the wrapper. Samtools was not listed in the first versions of the hicBuildMatrix wrapper, only since HiCExplorer version 1.5, with a commit in December 2016: https://github.com/deeptools/HiCExplorer/commit/d3b1279ef1819a7efa2ccc892e891a8a52f83c1a

**Comment by bgruening** [comment]
Nice find! Thanks a lot!

**Comment by mvdbeek** [comment]
Attention: deployment failure!

https://github.com/galaxyproject/tools-iuc/actions/runs/12184251422

**Comment by mvdbeek** [comment]
Attention: deployment failure!

https://github.com/galaxyproject/tools-iuc/actions/runs/12184251422

**Comment by joachimwolff** [comment]
```
Could not update suite_hicexplorer
Unexpected HTTP status code: 500: {"err_msg": "The repository_dependencies.xml file contains an invalid <repository> tag.  Unable to locate repository with name hicexplorer_hicbuildmatrixmicroc and owner bgruening.  "}
Failed to update repository contents for repository 'suite_hicexplorer' on the main Tool Shed.
Repository metadata updated successfully for repository 'suite_hicexplorer' on the main Tool Shed.
```

Did I forget to add the new tool hicBuildMatrixMicroC anywhere for the suite hicexplorer?

**Comment by bgruening** [comment]
I did it manually, we have the problem that those tools are under `bgruening` user for historical reasons and new tools ... ah complicated. I think I fixed that yesterday evening. It should be there now.

**Pull request merged**